backend/satellite_tools/reposync.py

```diff
@@ -1257,6 +1257,19 @@ def import_package_batch(self, to_process, to_disassociate, is_non_local_repo, b
                     # Set to_link to False, no need to link again
                     to_process[index] = (pack, True, False)
 
+            except KeyboardInterrupt:
+                raise
+            except rhnSQL.SQLError:
+                raise
+            except Exception:
+                failed_packages += 1
+                e = str(sys.exc_info()[1])
+                if e:
+                    log2(0, 1, e, stream=sys.stderr)
+                if self.fail:
+                    raise
+                to_process[index] = (pack, False, False)
+            finally:
```

Comment:
What happens if in the finally block an exception is thrown?
Especially on package import it can happen. Do we need to handle it?

Reply:
I see you point, but I think we don't actually need to handle that SQL exception at that point.

Before this PR, in case a SQL exception is produced while running `packageImport.PackageImport`, we were catching it but then let it to raise:

https://github.com/uyuni-project/uyuni/blob/master/backend/satellite_tools/reposync.py#L1283-L1284

So were getting an SQL exception when that happened.

Now, after this PR we're basically doing the same: we do not catch that exception but just let it to be raised directly.

So, when there is an SQL error at that time when running `packageImport.PackageImport` (which should not happen), the exception will be raised as we were doing before. 

If we catch that SQL exception and let reposync to continue, we know reposync will fail at "linking packages" because the entire batch was not imported.

Reply:
@meaksh I follow your argument and agree it holds for `SQLError`s (and, to be pedantic, also `KeyboardInterrupt`s).

Questions:
 1- how about `Exception`s different than the above two types and happening in `PackageImport.run`? Previous code logged an error, and either raised (if `--fail` was specified on command line) or marked the packet as not to be downloaded or linked and moved on. Current code would always fail, right? I am not necessarily saying the new behavior is less desirable - that all depends on what those `Exception`s could possibly be. Do you have any idea what could end up there?
2- in the previous version, the `finally` would clean up directories no matter whether any `Exception`s happened in any part of the code or not, and no matter whether those were handled or not. In the new version, if any `Exception` happens in `PackageImport.run` or surrounding code then cleanup code will be skipped. Is this really intended?

BTW: thanks for digging into this and come out alive (twice!)

I know the code is pretty labyrinthic and all my latest changes to add parallelism in the interest of speed also added to the complication somewhat.

:mechanical_arm: :mechanical_arm: :mechanical_arm: for @meaksh !